marketplace/deployer_util/provision.py

```diff
@@ -658,26 +667,32 @@ def provision_service_account(schema, prop, app_name, namespace,
   return sa_name, add_preprovisioned_labels(manifests, prop.name)
 
 
-def provision_storage_class(schema, prop, app_name, namespace):
-  if prop.storage_class.ssd:
-    sc_name = dns1123_name('{}-{}-{}'.format(namespace, app_name, prop.name))
-    manifests = [{
-        'apiVersion': 'storage.k8s.io/v1',
-        'kind': 'StorageClass',
-        'metadata': {
-            'name': sc_name,
-        },
-        # Some intelligence might go here to determine what
-        # provisioner and configuration to use here and below.
-        'provisioner': 'kubernetes.io/gce-pd',
-        'parameters': {
-            'type': 'pd-ssd',
-        }
-    }]
-    return sc_name, add_preprovisioned_labels(manifests, prop.name)
-  else:
-    raise Exception('Do not know how to provision for property {}'.format(
-        prop.name))
+def provision_storage_class(schema, prop, app_name, namespace, provisioner):
+  if not provisioner:
+    provisioner = _DEFAULT_STORAGE_CLASS_PROVISIONER
+
+  if provisioner == 'kubernetes.io/vsphere-volume':
+    parameters = {'diskformat': 'thin'}
+  elif provisioner == 'kubernetes.io/gce-pd':
+    if prop.storage_class.ssd:
+      parameters = {
+          'type': 'pd-ssd',
+      }
+    else:
+      raise Exception('Do not know how to provision for property {}'.format(
+          prop.name))
+
```

Comment:
Does there need to be an extra `else` in case the provisioner is set to another value?